### ISSUE TITLE
Can't set optional child fields

### ISSUE THREAD

**Issue by moshest** [neuledge/ddb-table#54]
TypeScript validation is failing when trying to sent an optional child field value. For example:

```ts
interface MyDoc {
  id: string;
  foo?: {
    bar: string;
  }
}

const table = new Table< MyDoc, 'id'>({
  tableName: 'MyDocs',
  primaryKey: 'id',
});

await table.update().set(['foo', 'bar'], 'hello');
``` 

`bar` key will mark as `never` since `MyDoc['foo']` is optional.

**Comment by jo-sip** [comment]
Just came across this issue while trying to find a way to update nested fields, is this the way to go about it? Would set("foo.bar", 'hello') also work?

**Comment by moshest** [comment]
I hope I will able to resolve this later this week..

In the meantime, you should be able to overcome this if you ignore the types:


```ts
await table.update().set(['foo', 'bar'] as never, 'hello' as never);
```

**Comment by moshest** [comment]
I'm not sure it's possible to support this without optionally breaking the types.

For example, in the following example an update will break the type definition and will create a type of `foo: { bar: string }` without the `baz` parameter within it. So the type error is in place.

```ts
interface MyDoc {
  id: string;
  foo?: {
    bar: string;
    baz: number;
  }
}

const table = new Table< MyDoc, 'id'>({
  tableName: 'MyDocs',
  primaryKey: 'id',
});

await table.update().set(['foo', 'bar'], 'hello');
```

I'm not sure what's the best way to handle this, I will keep the issue open for now.

**Comment by jo-sip** [comment]
Thanks for looking into this. I actually ran into a problem with Dynamo, where you can't actually update a nested attribute if the parent object doesn't exist. In your example, foo would already have to exist otherwise it'd fail, I assumed that it would just upsert foo, but that's not the case. A workaround is `try` update `catch` create pattern that I found works well.

So I don't think it'd necessarily break the types, it would just update the singular field `#foo.#bar` in this case, `baz` would already need to be created initially.

```
try {
    // Try update foo.
    await table.update.set(["foo", "bar"], "hello");
} catch (e) {
    // ValidationException
    await table.update.set("foo", {"bar" :"hello", "baz" : ""});
}
```


https://thomasstep.com/blog/handling-dynamodb-validation-exception-for-maps

https://stackoverflow.com/questions/35377729/how-to-set-a-dynamodb-map-property-value-when-the-map-doesnt-exist-yet

**Comment by moshest** [comment]
Fix released at `v2.0.1` 🎉
